Insomnia: After-response script example (#144)
Kong/developer.konghq.com: app/_how-tos/set-a-value-from-a-response-as-an-environment-variable.md

---

## 1. Configure the request

In this example, we want to generate a token for a {{site.konnect_short_name}} system account and set it as an environment variable. We first need to configure the [Create System Account Access Token](https://docs.konghq.com/konnect/api/identity-management/latest/#/System%20Accounts%20-%20Access%20Tokens/post-system-accounts-id-access-tokens) request to generate the token.

If you imported the [Konnect Identity Management API](https://docs.konghq.com/konnect/api/identity-management/latest/) specification into a new collection, most of the parameters are already set.

1. Click **Base Environment**, then click the **OpenAPI env global.api.konghq.com** sub environment to use its variables.
1. Click the pencil icon to update the environment if needed. For this example, you need to:
  * Update the `bearerToken` variable to your personal access token.
  * Create an `accountId` variable with the system account ID.
1. Open the **Create System Account Access Token** request, go to the **Body** tab, and update the token's name and expiration date and time.

## 2. Create the after-response script

1. Open the **Scripts** tab and click **After-response**.

> 1. Add the following script. It will check that the request returns the expected 201 status code, get the token value, and set it as the value of a new `systemToken` environment variable.

Reviewer: ```suggestion
1. Add the following script. When this script run it will: 
    * Check that the request returns the expected 201 status code
    * Get the value of the token
    * Assign the value to the `systemToken` environment variable.
```
